src/main/java/unicash/ui/CommandBox.java

```diff
@@ -83,4 +123,58 @@ public interface CommandExecutor {
         CommandResult execute(String commandText) throws CommandException, ParseException;
     }
 
+    /**
+     * Adds the input string to command history list and resets command index.
+     */
+    private void addUserInputToHistory(String input) {
+        userInputHistory.add(input);
+        currentUserInputIndex = -1; // reset index as new command is added
+    }
+
+    /**
+     * Clears the Command Text Field.
+     */
+    private void clearCommandTextField() {
+        commandTextField.setText("");
+    }
+
+    /**
+     * Iterates through userInput history backwards and sets the commandTextField text
+     * stopping at the earliest entered user input.
+     */
+    private void showPreviousUserInput() {
+        if (userInputHistory.isEmpty()) {
+            return;
+        }
+
+        if (currentUserInputIndex == -1) {
```

Comment:
See comment above

Reply:
Fixed as well, changed this to refer to the class constant